Assets/MixedRealityToolkit.SDK/Experimental/Features/Support/Scripts/BaseServiceManager.cs

```diff
@@ -0,0 +1,395 @@
+#else
+            if (!Application.platform.IsPlatformSupported(supportedPlatforms))
+#endif
+            {
+                return default(T);
+            }
+
+            if (!typeof(T).IsAssignableFrom(concreteType))
+            {
+                Debug.LogError($"Error: {concreteType.Name} service must implement {typeof(T)}.");
+                return default(T);
+            }
+
+            try
+            {
+                T serviceInstance = (T)Activator.CreateInstance(concreteType, args);
+                return serviceInstance;
+            }
+            catch (Exception e)
+            {
+                Debug.LogError($"Error: Failed to instantiate {concreteType.Name}: {e.GetType()} - {e.Message}");
+                return default(T);
+            }
+        }
+
+        #endregion IMixedRealityServiceRegistrar implementation
+
+        /// <summary>
+        /// Initialize a service.
+        /// </summary>
+        /// <typeparam name="T">The interface type for the service to be intialized.</typeparam>
+        /// <param name="concreteType">The concrete type of the service to initialize.</param>
+        /// <param name="supportedPlatforms">The platoform(s) on which the service is supported.</param>
+        /// <param name="args">Arguments to provide to the service class constructor.</param>
+        protected virtual void Initialize<T>(Type concreteType, SupportedPlatforms supportedPlatforms = (SupportedPlatforms)(-1), params object[] args) where T : IMixedRealityService
+        {
+            RegisterService<T>(concreteType, supportedPlatforms, args);
+            T serviceInstance = FindService<T>();
+            serviceInstance?.Initialize();
+        }
+
+        /// <summary>
+        /// Uninitialize a service.
+        /// </summary>
+        /// <typeparam name="T">The interface type for the service to uninitialize.</typeparam>
+        protected virtual void Uninitialize<T>() where T : IMixedRealityService
+        {
+            T serviceInstance = FindService<T>();
+
+            if (serviceInstance != null)
+            {
+                registeredServices.Remove(typeof(T));
+                MixedRealityServiceRegistry.RemoveService<T>(serviceInstance, this);
+            }
+        }
+
+        /// <summary>
+        /// Locates a service instance in the registry,
+        /// </summary>
+        /// <typeparam name="T">The interface type of the service to locate.</typeparam>
+        /// <param name="name">The name of the desired service.</param>
+        /// <returns>Instance of the interface type, or null if not found.</returns>
+        private T FindService<T>(string name = null) where T : IMixedRealityService
+        {
+            Type interfaceType = typeof(T);
+
+            if (!registeredServices.ContainsKey(interfaceType)) { return default(T); }
+
+            return (T)registeredServices[interfaceType];
```

Comment:
Use [https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2.trygetvalue?view=netframework-4.8](https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2.trygetvalue?view=netframework-4.8) instead of doing two lookups.

Reply:
fixed